File: integrations/runtime-bridges/go/bridge.go
package runtimebridges

type KernelBridge struct {
	Config map[string]any
}

func (k *KernelBridge) Init(config map[string]any) {
	k.Config = config
}

func (k *KernelBridge) Invoke(action map[string]any) map[string]any {
	return map[string]any{"status": "todo", "action": action}
}

func (k *KernelBridge) Stream() <-chan map[string]any {
	ch := make(chan map[string]any, 1)
	ch <- map[string]any{"status": "todo", "event": "stream-not-implemented"}
	close(ch)
	return ch
}

func (k *KernelBridge) Close() {}

File: integrations/runtime-bridges/rust/bridge.rs
use std::collections::HashMap;

pub struct KernelBridge {
    config: HashMap<String, String>,
}

impl KernelBridge {
    pub fn new() -> Self {
        Self { config: HashMap::new() }
    }

    pub fn init(&mut self, config: HashMap<String, String>) {
        self.config = config;
    }

    pub fn invoke(&self, action: HashMap<String, String>) -> HashMap<String, String> {
        let mut out = HashMap::new();
        out.insert("status".into(), "todo".into());
        out.insert("action".into(), format!("{:?}", action));
        out
    }

    pub fn stream(&self) -> HashMap<String, String> {
        let mut out = HashMap::new();
        out.insert("status".into(), "todo".into());
        out.insert("event".into(), "stream-not-implemented".into());
        out
    }

    pub fn close(&self) {}
}
